src/libraries/System.Private.CoreLib/src/System/Threading/Thread.Windows.cs

```diff
@@ -5,7 +5,10 @@ namespace System.Threading
 {
     public sealed partial class Thread
     {
-        private static Exception GetApartmentStateChangeFailedException() =>
-            new InvalidOperationException(SR.Thread_ApartmentState_ChangeFailed);
+        private static Exception GetApartmentStateChangeFailedException()
+        {
+            string msg = SR.Format(SR.Thread_ApartmentState_ChangeFailed, Thread.CurrentThread.GetApartmentState());
```

Comment:
It may better to add `throwOnError` argument to `TrySetApartmentStateUnchecked` and throw the exception right from there. The current apartment state is available in that function already. No need to re-fetch it.